pkg/app/piped/executor/analysis/analysis.go

```diff
@@ -96,6 +100,31 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 
 	eg, ctxWithTimeout := errgroup.WithContext(ctxWithTimeout)
 
+	// Sync the skip command.
+	var (
+		status = model.StageStatus_STAGE_NOT_STARTED_YET
+		doneCh = make(chan struct{})
+	)
+	defer close(doneCh)
+	go func() {
+		ticker := time.NewTicker(5 * time.Second)
+		defer ticker.Stop()
```

Comment:
Since the `errgroup` is being used so how about running the checker inside a new goroutine of that errgroup instead of using a `raw` goroutine + `doneCh`?
https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go

the checker can return a "skipError"  to check at `Wait` function.

Reply:
I'm sorry if my recognition is wrong.
I dare to generate a goroutine separately from the `errgroup` since the goroutine generated by the `errgroup` waits until all routines are completed.

Reply:
https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go
> The first call to return a non-nil error cancels the group; its error will be returned by Wait.

And the `Wait` waits for the completion of all its goroutines.
But I think that is what we are expecting.
Once it got a "SKIP" command and the executing goroutines receive the cancel signal and stop their work.
And then the stage waits for the canceling of all its analyses before returning a completion status.

Reply:
We can do that when we received the skipped command but I wonder how to handle simply whether the all of analyses were already done or not in order to shutdown the goroutine handing the skip command gracefully.

Reply:
I got your point. It would be complex to cancel the goroutine which is handling the command checker once all analyses are completed normally.

Reply:
I got your point. It would be complex to cancel the goroutine which is handling the command checker once all analyses are completed normally.